src/raftstore/store/snap.rs

```diff
@@ -1009,6 +1016,32 @@ impl SnapManager {
         SnapManagerBuilder::default().build(path, router)
     }
 
+    fn gen_migrate_target(origin_filename: &str) -> (String, String) {
+        // Old snapshots files are named like
+        // "type_region_term_index.meta" or
+        // "type_region_term_index_cf.sst",
+        // where "type" is one of "gen" or "rev".
+        // Now "type_region_term_index" is the folder name
+        // and "meta" or "cf.sst" is the filename
+        let parts: Vec<_> = origin_filename.splitn(2, '.').collect();
+        let components: Vec<_> = parts
+            .first()
+            .map_or(Default::default(), |base| base.splitn(5, '_').collect());
+        let (snapname, basename) = if components.len() >= 5 {
+            (&components[..4], components[4])
+        } else {
+            (&components[..], Default::default())
+        };
+        let dirname = snapname.join(&"_");
+        let extname = parts.get(1).copied().unwrap_or_default();
+        let filename = if basename.is_empty() {
+            String::from(extname)
+        } else {
+            basename.to_owned() + "." + extname
+        };
```

Comment:
ugly implementation, i prefer
if is mata file {
...
} else sst file {
...
}

Reply:
I change the condition to checking the filename ends with ".meta"

Reply:
Still hard to read.

Reply:
Still hard to read. Please refine this function.